bin/sync/article_updater.rb

```
@@ -0,0 +1,155 @@
+      end
+
+    rescue StandardError => e
+      if attempt < max_retries
+        puts "Error updating canonical_url for article #{article_id}, retrying in #{attempt * 2} seconds..."
+        sleep(attempt * 2)
+        retry
+      else
+        puts "Failed to update canonical_url for article #{article_id} after #{max_retries} attempts: #{e.message}"
+      end
+    end
+  end
```

rubocop

[convention] Assignment Branch Condition size for update_canonical_url_on_dev_to is too high. [<10, 20, 6> 23.15/23]

(Metrics/AbcSize)